modules/redpanda/options.go

```diff
@@ -128,6 +133,21 @@ func WithEnableSchemaRegistryHTTPBasicAuth() Option {
 	}
 }
 
+// WithEnableHTTPProxyBasicAuth enables HTTP basic authentication for
+// HTTP Proxy (pandaproxy).
+func WithEnableHTTPProxyBasicAuth() Option {
+	return func(o *options) {
+		o.HTTPProxyAuthenticationMethod = "http_basic"
+	}
+}
+
+// WithHTTPProxyOIDCAuth enables OIDC authentication for HTTP Proxy (pandaproxy).
+func WithHTTPProxyOIDCAuth() Option {
+	return func(o *options) {
+		o.HTTPProxyAuthenticationMethod = "oidc"
+	}
+}
```

Comment:
question: given the values for the HTTP Proxy authentication method are opposite (none, basic or oidc), this API could be a bit confusing as it allows passing multiple options with values that could cause conflict:

```go
ctr, err := redpanda.Run(ctx, img, redpanda.WithEnableHTTPProxyBasicAuth(), redpanda.WithHTTPProxyOIDCAuth(), opts...)
```

Do you think we could have one single option, like `WithHTTPProxyAuthMethod(method HTTPProxyAuthMethod)`? Please check I'm proposing a type for the methods, to avoid strings (plus validation). If this option is not passed, then "none" will be applied (already done in this PR as part of the defaults).